packages/fetchai/connections/p2p_libp2p/aea/pipe_windows.go

```diff
@@ -0,0 +1,81 @@
+// +build windows !linux !darwin
+
+/* -*- coding: utf-8 -*-
+* ------------------------------------------------------------------------------
+*
+*   Copyright 2018-2019 Fetch.AI Limited
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*       http://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*
+* ------------------------------------------------------------------------------
+ */
+
+package aea
+
+import (
+	"encoding/binary"
+	"net"
+	"strconv"
+)
+
+type TCPSocketPipe struct {
+	port uint16
+	conn net.Conn
+}
+
+func (sock *TCPSocketPipe) Connect() error {
+	// open tcp connection
+	var err error
+	sock.conn, err = net.Dial("tcp", "127.0.0.1:"+strconv.FormatInt(int64(sock.port), 10))
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (sock *TCPSocketPipe) Read() ([]byte, error) {
+	// TOFIX(LR) duplicated code to avoid circular dep
+	//           utils.ReadBytesConn(sock.conn)
+	buf := make([]byte, 4)
+	_, err := sock.conn.Read(buf)
+	if err != nil {
+		return buf, err
+	}
+	size := binary.BigEndian.Uint32(buf)
+
+	buf = make([]byte, size)
+	_, err = sock.conn.Read(buf)
+	return buf, err
+}
+
+func (sock *TCPSocketPipe) Write(data []byte) error {
+	// TOFIX(LR) duplicated code to avoid circular dep
+	//    		 utils.WriteBytesConn(sock.conn, data)
+	size := uint32(len(data))
+	buf := make([]byte, 4, 4+size)
```

Comment:
Why here `buf := make([]byte, 4, 4+size)` and for UnixPipe Write: `buf := make([]byte, 4)`?

Reply:
well spotted, UnixPipe should be one write only as well